core/src/test/scala/com/nvidia/spark/rapids/tool/tuning/BaseAutoTunerSuite.scala

```diff
@@ -171,12 +171,29 @@ abstract class BaseAutoTunerSuite extends FunSuite with BeforeAndAfterEach with
    * expected results.
    */
   protected def compareOutput(expectedResults: String, autoTunerOutput: String): Unit = {
-    val outputsMatch = expectedResults == autoTunerOutput
-    assert(outputsMatch,
-      s"""|=== Expected ===
+    val expectedLines = expectedResults.split("\n").toSeq
+    val actualLines = autoTunerOutput.split("\n").toSeq
+    val maxLines = Math.max(expectedLines.length, actualLines.length)
+    var hasMismatch = false
+    val mismatches = new StringBuilder()
+    for (i <- 0 until maxLines) {
+      val expectedLine = if (i < expectedLines.length) expectedLines(i) else ""
+      val actualLine = if (i < actualLines.length) actualLines(i) else ""
+      if (expectedLine != actualLine) {
+        hasMismatch = true
+        mismatches.append(s"Line ${i + 1}:\n")
+        mismatches.append(s"  Expected: $expectedLine\n")
+        mismatches.append(s"  Actual  : $actualLine\n")
+      }
+    }
+    assert(!hasMismatch,
+      s"""|=== Mismatched Lines ===
+          |$mismatches
```

Comment:
This helps me to find mismatch more quickly.

Reply:
Yeah, mainly to help me find the actual position. I had cases that an extra empty line is inserted mistakenly and I copy paste both expect and actual output to a text compare website to check and the website told me they are exactly the same as I totally missed the empty line when copy paste :(. Finally I had to add this piece of code to help...